classes/user/maps/Schema.php

```diff
@@ -72,36 +81,112 @@ public function summarizeReviewer(User $item, array $auxiliaryData = []): array
+            'reviewsActive','reviewsCompleted','reviewsDeclined','reviewsCancelled',
+            'averageReviewCompletionDays','dateLastReviewAssignment','reviewerRating',
+        ]);
     }
 
     /**
      * Summarize a collection of users
      *
      * @see self::summarize
      */
-    public function summarizeMany(Enumerable $collection): Enumerable
+    public function summarizeMany(Enumerable $collection, array $options = []): Enumerable
     {
-        $this->collection = $collection;
-        return $collection->map(function ($item) {
-            return $this->summarize($item);
-        });
+        return $this->mapUsersWithPermissions($collection, $this->getSummaryProps(), $options);
     }
 
     /**
      * Summarize a collection of reviewers
      *
      * @see self::summarizeReviewer
      */
-    public function summarizeManyReviewers(Enumerable $collection): Enumerable
+    public function summarizeManyReviewers(Enumerable $collection, array $options = []): Enumerable
     {
-        $this->collection = $collection;
-        return $collection->map(function ($item) {
-            return $this->summarizeReviewer($item);
-        });
+        return $this->mapUsersWithPermissions($collection, $this->getReviewerSummaryProps(), $options);
+    }
+
+    /**
+     * Shared implementation for mapping/summarizing collections
+     */
+    private function mapUsersWithPermissions(Enumerable $collection, array $props, array $options): Enumerable
+    {
+        // handles lazy collection
+        $users = collect($collection->all());
+
+        $this->collection = $users;
+
+        $userIds = $users
+            ->map(fn (User $u) => (int) $u->getId())
+            ->unique()
+            ->values()
+            ->all();
+
+        $currentUserId = $options['currentUserId'] ?? null;
+        $isSiteAdmin   = (bool)($options['isSiteAdmin'] ?? Validation::isSiteAdmin());
+        $options['isSiteAdmin'] = $isSiteAdmin;
+
+        if (empty($userIds)) {
+            $options['permissionMap'] = [];
+        } elseif ($isSiteAdmin) {
+            $options['permissionMap'] = array_fill_keys($userIds, true);
+        } elseif ($currentUserId !== null) {
+            $options['permissionMap'] = $this->buildUserContextManagementMap((int)$currentUserId, $userIds);
+        } else {
+            $options['permissionMap'] = array_fill_keys($userIds, false);
+        }
+
+        return $users->map(fn (User $u) => $this->mapByProperties($props, $u, $options));
+    }
+
+
+    /**
+     * build a permission map for a manager over a set of users
+     */
+    private function buildUserContextManagementMap(int $managerUserId, array $managedUserIds): array
+    {
+        if (empty($managedUserIds)) {
+            return [];
+        }
+
+        $rows = DB::table('users as u')
+            ->select('u.user_id')
+            ->selectRaw(
+                'CASE WHEN EXISTS (
+                    SELECT 1
```

Comment:
Is it correct? At the first glance it should select something like `context_id` column